test/functional/operatingsystems_controller_test.rb

```diff
@@ -1,8 +1,12 @@
 require 'test_helper'
 class OperatingsystemsControllerTest < ActionController::TestCase
-  def setup_user
-    @request.session[:user] = users(:one).id
-    users(:one).roles       = [Role.default, Role.find_by_name('Viewer')]
+  def setup_user(operation = nil, type = 'operatingsystems')
+    if operation.nil?
```

Comment:
It might be better to rename this OS-specific helper so it doesn't override the generic one - it'll be easier to see where it's used and remove it in future that way.